cloud/rackspace/rax_facts.py

```diff
@@ -98,6 +98,11 @@ def rax_facts(module, address, name, server_id):
         except Exception, e:
             pass
 
+    #rackspace likes to leave servers in the deleted state for a while.  
+    #If you make a server with the same name it triggers len(servers) > 1 
+    #so just remove servers with a server status of deleted from the list
+    servers = [server for server in servers if server.status != "DELETED"]
```

Comment:
Everywhere I am reading suggests using a list comprehension as opposed to iteration and then deleting (in this scenario you are iterating the list twice)

http://stackoverflow.com/questions/1207406/remove-items-from-a-list-while-iterating-in-python

Based on that I would recommend using a slice to avoid duplicating the list.  I could also just change the variable name if you want.